en/peripherals/uavcan_escs.md

```diff
@@ -28,52 +31,53 @@ Sapog-based ESCs:
 
-## Sapog-based ESC setup {#sapog}
-
-The following sections explain how to use [Sapog](https://github.com/PX4/sapog#px4-sapog)-based ESCs with PX4.
-The instructions should work for any Sapog-based ESC design.
 
-### Connecting {#connecting}
+## Wiring/Connections {#connecting}
 
-Connect all of the on-board CAN devices into a chain and make sure the bus is terminated at the end nodes.
+Connect all of the on-board UAVCAN devices into a chain and make sure the bus is terminated at the end nodes.
 The order in which the ESC are connected/chained does not matter.
+
 For more information information about proper bus connections see [UAVCAN Device Interconnection](https://kb.zubax.com/display/MAINKB/UAVCAN+device+interconnection) (Zubax KB).
 
-### Configuring Sapog
 
-PX4 supports automatic ESC enumeration, so no manual configuration is needed.
-During the auto-enumeration process, the following configuration parameters will be assigned automatically for each enumerated ESC:
-- `esc_index`
-- `ctl_dir`
+## PX4 Configuration
 
-Learn more about configuration parameters in the [Sapog reference manual](https://files.zubax.com/products/io.px4.sapog/Sapog_v2_Reference_Manual.pdf).
+In order to use a UAVCAN ESC with PX4 you will need to enable the UAVCAN driver (and you may need to [set other parameters](../advanced_config/parameters.md)):
+1. Power the vehicle using the battery (you must power the whole vehicle, not just the flight controller!) and connect *QGroundControl*.
+1. Navigate to the **Vehicle Setup > Parameters** screen.
+1. Set [UAVCAN_ENABLE](../advanced_config/parameter_reference.md#UAVCAN_ENABLE) to the value *Sensors and Motors* (3) and then reboot the flight controller.
+1. If applicable, set [UAVCAN_ESC_IDLT](../advanced_config/parameter_reference.md#UAVCAN_ESC_IDLT) to 1 in order to ensure that the motors are always running at least at the idle throttle while the system is armed.
+   Some systems will not benefit from this behavior, e.g. glider drones.
 
-> **Tip** You can skip the automated enumeration procedure by configuring the above listed parameters manually on each ESC using the [UAVCAN GUI Tool](https://uavcan.org/GUI_Tool/Overview/).
 
-<span></span>
-> **Note** You may need to perform some tuning of motor control characteristics.
-  That is "out of the scope" and may be done later.
+## ESC Setup
 
-### Configuring PX4
+While UAVCAN devices are generally *plug'n'play* you will still need to enumerate (number) each of the ESC used in your system and set their direction so that they can be identified/controlled by PX4.
 
-Power the vehicle using the battery (you must power the whole vehicle, not just the flight controller!) and connect *QGroundControl*.
-Then navigate to the [Vehicle Setup > Parameters](../advanced_config/parameters.md) screen.
+> **Note** The ESC index and direction must match/map to the [Airframe Reference](../airframes/airframe_reference.md) for the vehicle type.
```

Comment:
@bkueng @pavel-kirienko This is my GUESS on how PX4 understands/maps mixer definitions of motor order to ESC. Is it correct?

Reply:
@pavel-kirienko Any thoughts on this?

Reply:
Your guess seems correct.

Reply:
Thanks @pavel-kirienko. @cmic0 FYI if you know different let me know!